src/ProgressOnderwijsUtils/SingleSignOn/SsoProcessor.cs

```diff
@@ -88,28 +88,34 @@ static string EncodeQueryParameter(string key, string value)
                 return Maybe.Error("Missing attribute " + UID);
             }
 
+            var (inresponseTo, notOnOrAfter) = GetSubjectConfirmationData(assertion);
+
+            var now = DateTime.UtcNow;
+            if (now >= notOnOrAfter) {
+                return Maybe.Error($"Expired: {now} >= {notOnOrAfter}");
+            }
+
             return Maybe.Ok(
                 new SsoAttributes {
                     uid = uid,
                     domain = GetNullableAttribute(assertion, DOMAIN),
                     email = GetAttributes(assertion, MAIL),
                     roles = GetAttributes(assertion, ROLE),
-                    InResponseTo = GetInResponseTo(assertion),
-                    IssueInstant = (DateTime)assertion.Attribute("IssueInstant"),
+                    InResponseTo = inresponseTo,
                     AuthnContextClassRef = (string)authnStatement.Element(SamlNamespaces.SAML_NS + "AuthnContext").Element(SamlNamespaces.SAML_NS + "AuthnContextClassRef"),
                 });
         }
 
-        static string? GetInResponseTo(XElement assertion)
+        static (string? inresponseTo, DateTime notOnOrAfter) GetSubjectConfirmationData(XElement assertion)
         {
-            // ReSharper disable PossibleNullReferenceException
-            var rawInResponseTo = (string)assertion
-                .Element(SamlNamespaces.SAML_NS + "Subject")
-                .Element(SamlNamespaces.SAML_NS + "SubjectConfirmation")
-                .Element(SamlNamespaces.SAML_NS + "SubjectConfirmationData")
-                .Attribute("InResponseTo");
-            // ReSharper restore PossibleNullReferenceException
-            return XmlConvert.DecodeName(rawInResponseTo);
+            var subjectConfirmationData = assertion
+                    .Element(SamlNamespaces.SAML_NS + "Subject")
+                    .Element(SamlNamespaces.SAML_NS + "SubjectConfirmation")
+                    .Element(SamlNamespaces.SAML_NS + "SubjectConfirmationData");
+            return (
+                XmlConvert.DecodeName((string?)subjectConfirmationData.Attribute("InResponseTo")),
+                (DateTime)subjectConfirmationData.Attribute("NotOnOrAfter")

```

Comment:
Wat is de DateTimeKind hier?  want je doet `>=` hoger op, en dat doet gestoorde dingen met kind mismatches.

Reply:
In de responses die ik heb gezien is het universal.

Reply:
spec utc: https://www.oasis-open.org/committees/download.php/35711/sstc-saml-core-errata-2.0-wd-06-diff.pdf 

...maar niet in de xsd, dus een assert is niet gestoord.

Reply:
(geen idee of wij een of andere xsd checken trouwen, I just googled it)